### PR TITLE
Stats: Update module-followers and module-followers-page to updated component directory

### PR DESCRIPTION
**Pull request by alternatekev**
This is a re-do of #2171, which moves the `stats/module-followers.jsx` and `stats/module-followers-page.jsx` to `stats/stats-followers/index.jsx` and `stats/stats-followers/followers-page.jsx`

This PR fixes the weird problems I introduced in #2171 with insights.jsx and follows.jsx being added anew.

ping @dmsnell and @timmyc

### CODE REVIEW

**Comment by dmsnell** [comment]
@alternatekev is the `StatsFollowersPage` module a replacement for the `StatModuleFollowers` module?

**Comment by alternatekev** [comment]
@dmsnell No, it replaces `module-followers-page.jsx`

**Comment by dmsnell** [comment]
@alternatekev it looks like we have two components here that end up with the same `DisplayName`

 - `/stats/stats-followers/followers-page.jsx`
 - `client/my-sites/stats/stats-followers/index.jsx`

**Comment by alternatekev** [comment]
@dmsnell fixed!

File: client/my-sites/stats/stats-followers/index.jsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+			activeFilter: 'wpcom-followers',
+			noData: this.props.followersList.isEmpty( 'subscribers' )
+		};
+	},
+
+	componentWillReceiveProps( nextProps ) {
+		this.setState( {
+			noData: nextProps.followersList.isEmpty( 'subscribers' )
+		} );
+	},
```
Comment (dmsnell) [inline client/my-sites/stats/stats-followers/index.jsx]: This seems a bit dubious to me since I can't find anywhere that we actually use this state value. In `render()` it uses the `isEmpty` function call directly, which makes more sense to me than storing it in `state`

File: client/my-sites/stats/stats-followers/index.jsx [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+	recordDownloadClick() {
+		analytics.ga.recordEvent( 'Stats', 'Clicked Download Email Followers CSV link' );
+	},
+
+	render() {
+		const data = this.data();
+		const hasError = this.props.followersList.isError();
+		const noData = this.props.followersList.isEmpty( 'subscribers' );
+		const infoIcon = this.state.showInfo ? 'info' : 'info-outline';
```
Comment (dmsnell) [inline client/my-sites/stats/stats-followers/index.jsx]: with as many references we have in this function to `this.props.` it might be handy to use some destructuring assignment.

```js
const { followList followersList, followType, page, pageClick, perPage } = this.props;
const hasError = followersList.isError();
const noData = followersList.isEmpty( 'subscribers' );

…

switch ( followType ) {
	…
}

// etc…
```

File: client/my-sites/stats/stats-followers/index.jsx [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+				break;
+
+			case 'email':
+				itemType = this.translate( 'Email' );
+				break;
+
+			case 'wpcom':
+				itemType = this.translate( 'WordPress.com' );
+				break;
+		}
```
Comment (dmsnell) [inline client/my-sites/stats/stats-followers/index.jsx]: with a basic lookup like this, it might be more concise both as an object with key lookup and as a separate function…

```js
const followTypeName = type => {
	const typeNames = {
		'comment': this.translate( 'Comments' ),
		'email': this.translate( 'Email' ),
		'wpcom': this.translate( 'WordPress.com' )
	};

	return typeNames[ type ];
}
```

File: client/my-sites/stats/stats-followers/index.jsx [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+		if ( data.total ) {
+			startIndex = this.props.perPage * ( this.props.page - 1 ) + 1;
+			endIndex = this.props.perPage * this.props.page;
+
+			if ( endIndex > data.total ) {
+				endIndex = data.total;
+			}
```
Comment (dmsnell) [inline client/my-sites/stats/stats-followers/index.jsx]: how about folding this into the the above statement?

```js
endIndex = Math.min( data.total, page * perPage );
```

File: client/my-sites/stats/stats-followers/index.jsx [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+			paginationSummary = this.translate( 'Showing %(startIndex)s - %(endIndex)s of %(total)s %(itemType)s followers', {
+				context: 'pagination',
+				comment: '"Showing [start index] - [end index] of [total] [item]" Example: Showing 21 - 40 of 300 WordPress.com followers',
```
Comment (dmsnell) [inline client/my-sites/stats/stats-followers/index.jsx]: I like the example string here, but not sure about the value of the first half of this comment since it's basically a copy of the actual translation string.

File: client/my-sites/stats/stats-followers/index.jsx [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+					</ul>
+				</div>
+			);
+		}
+
+		return (
+			<Card className={ classNames.apply( null, classes ) }>
+				<div className="followers">
+					<div className="module-header">
+						{ moduleHeaderTitle }
```
Comment (dmsnell) [inline client/my-sites/stats/stats-followers/index.jsx]: Is there a reason not to simply include the content of this element here verses pulling it in as a separate variable? I think it would be easier to see what's being rendered if the `<h4 />` were inserted at this point.

**Comment by dmsnell** [comment]
@alternatekev the `<StatsFollowers />` has a really big `render` function. I think that it can be a bit hard to follow because of that. It also is a composition of lots of includes, like `{ pagination }`. In these cases we usually ask if we couldn't break apart the component into some separate React components, then instead of `{ pagination }` we could see directly what's there, such as 

```
<Pagination {...{ page, perPage, total, pageClick } />
```

(if you had already set those variables)

I think there's some more things like this in here that could be separated out to make a simpler read of that module.

**Comment by alternatekev** [comment]
Hi, @dmsnell, most of the things you're calling out have existed in this file since before it was moved. Can we stash these comments for after this particular PR? This PR is for moving the files and updating to ES6 only. I understand that there are lots of code issues here but we're trying to get some structural work done on the components before we get to the maintenance.

**Comment by dmsnell** [comment]
> Can we stash these comments for after this particular PR?

sure @alternatekev - I thought this was a new file. On a related note, it looks like we are combining file moves with ES6 syntax conversions, which has made it more difficult for me to review and figure out what's actually changing. This is also the reason I thought this was a new file.

Is there a way we could separate the two, such as making the file moves/renames a single PR and then do the syntax/formatting update in another PR?

**Comment by timmyc** [comment]
@dmsnell the combining of the ES6 + moves is really my fault as I started to do it when we began this maintenance iteration on stats.  I can see how it definitely makes the diff harder to follow so I apologize for that.

@alternatekev let me pull this branch down and give it a test

**Comment by timmyc** [comment]
Looks like this had some lingering `git mv` issues too.  After talking it over with @alternatekev I'm going to start from scratch on this one.
